### PR TITLE
test(RBAC): RHICOMPL-3017 test API routes are covered with RBAC enforcement

### PR DESCRIPTION
**Pull request by romanblanco**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
1c48c762-7b09-4a1a-8a49-e9a1f8484f2c
6a04c792-7a5e-46aa-93f6-8315b6755350

File: test/integration/routes_test.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class RoutesTest < ActionController::TestCase
+  test 'APIv1 routes are covered with RBAC enforcement' do
```
Comment (skateman) [inline test/integration/routes_test.rb:6]: Let's create a separate test case for each route, with controllers as contexts please. This way we would have a failure for each route if multiple ones are failing.

**Review by skateman — APPROVED** [review]
The Seal of Approval

File: test/integration/routes_test.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+require 'test_helper'
+
+class RoutesTest < ActionController::TestCase
+  controllers = Rails.application
+                     .routes.routes
+                     .map { |route| route.defaults[:controller] }
+                     .grep(%r{^v1/(?!statuses)}) # filter out undesired endpoints
+                     .uniq
+                     .map { |url| "#{url.classify.pluralize}Controller".constantize }
```
[the remark this answers is not in the document]
Comment (romanblanco) [reply test/integration/routes_test.rb:11]: Nope, I need to pluralize the classified and than add the `Controller`

[the remark this answers is not in the document]
Comment (romanblanco) [reply test/integration/routes_test.rb]: Fixed.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**